spec/services/workflow_errors_reporter_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.describe WorkflowErrorsReporter do
+  let(:put) { 'https://sul-lyberservices-test.stanford.edu/workflow/dor/objects/druid:jj925bx9565/workflows/preservationWF/moab-valid' }
+  let(:headers) { { 'Content-Type' => 'application/xml' } }
+
+  describe '.update_workflow' do
+    let(:error_msg) do
+      { 13 => "PreservedObjectHandler(jj925bx9565, 9, 6570668, <Endpoint:...>) Invalid moab, validation error...ential version directories. Current directories: [\\\"v0001\\\", \\\"v0002\\\", \\\"v0009\\\"]\"]" }
+    end
+
+    let(:body) { "<process name='moab-valid' status='error' errorMessage='#{error_msg}'/>" }
+
+    before do
+      Settings.workflow_services_url = 'https://sul-lyberservices-test.stanford.edu/workflow/'
+      stub_request(:put, put)
+        .with(body: body,
+              headers: headers)
+        .to_return(status: 204, body: "", headers: {})
+    end
+    it 'updates preservationWF with invalid moab errors' do
+      expect(described_class.update_workflow('jj925bx9565', 'moab-valid', error_msg)).to be_an_instance_of(Faraday::Response)
+    end
+
+    it 'rescues for Faraday::Error' do
+      allow(described_class).to receive(:update_workflow).and_raise(Faraday::Error)
+      expect { described_class.update_workflow('jj925bx9565', 'moab-valid', 'Foo() error') }.to raise_error(Faraday::Error)
+    end
+  end
+
+  describe '.update_workflow with invalid workflow_services.url' do
+    let(:body) { "<process name='moab-valid' status='error' errorMessage='[{13=>\"Foo() Invalid Moab\"}]'/>" }
+
+    before do
+      stub_request(:put, put)
+        .with(body: body,
+              headers: headers)
+        .to_return(status: 200, body: "", headers: {})
+    end
+    it 'returns Rails warning' do
+      Settings.workflow_services_url = ''
+      expect(Rails.logger).to receive(:warn).with('no workflow hookup - assume you are in test or dev environment')
+      described_class.update_workflow('jj925bx9565', 'moab-valid', [{ 13 => "Foo() Invalid Moab" }])
+    end
+  end
```

Comment:
do you want a test for a bad process name?  Or is that silly?   What about when a druid isn't recognized by workflows (like we will have in stage) ?